File: frontend/src/app/modules/tasker/models/tasker.model.ts
import {TaskerServiceModel} from "../../tasker-service/models/tasker-service.model";
import {RatingDetailModel} from "src/app/shared/models/rating-detail.model";
import {TasklionAccountModel} from "../../tasklion-account/models/tasklion-account.model";
import {PersonalDetailModel} from "../../tasklion-account/models/personal-detail.model";
import {ServiceAreaModel} from "../../tasker-service/service-area/models/service-area.model";

export interface TaskerModel {

  tasklionAccount: TasklionAccountModel;
  personalDetail: PersonalDetailModel;
  aboutMe?: string;
  serviceAreas?: ServiceAreaModel[];
  services: TaskerServiceModel[];
  ratingDetail?: RatingDetailModel;

}

File: backend/src/main/java/com/tasklion/backend/features/tasklionUser/tasker/model/TaskerModel.java
package com.tasklion.backend.features.tasklionUser.tasker.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.tasklion.backend.features.personalDetail.PersonalDetailModel;
import com.tasklion.backend.features.taskerService.model.TaskerServiceModel;
import com.tasklion.backend.features.taskerService.serviceArea.model.ServiceAreaModel;
import com.tasklion.backend.features.taskerService.serviceReview.model.RatingDetailModel;
import com.tasklion.backend.features.tasklionAccount.model.TasklionAccountModel;
import com.tasklion.backend.features.tasklionUser.TasklionUserModel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Data
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
public class TaskerModel extends TasklionUserModel {

    private TasklionAccountModel tasklionAccount;
    private PersonalDetailModel personalDetail;

    private String aboutMe;
    private List<ServiceAreaModel> serviceAreas;
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private List<TaskerServiceModel> services;
    private RatingDetailModel ratingDetail;

}
